ddl/cluster.go

```diff
@@ -36,15 +41,12 @@ var pdScheduleKey = []string{
 	"replica-schedule-limit",
 }
 
-func closePDSchedule(job *model.Job) error {
-	if err := savePDSchedule(job); err != nil {
```

Comment:
I think still you should save `PDSchedule` before close. Saving PD configs in stage 1 may be too early?

Reply:
But if you close the pd schedule in stage 2, then tidb panic before update the job.Args to kv, it can't resume pd schedule after ddl job finished.

Reply:
You can decode `job.Args` per stage, it is completely valid.

Reply:
Let image a statuation,

Closed `PDSchedule` and save the previous value in `job.Args`.  Now someone killed tidb-server before update `job.RawArgs`. After tidb-server restart, the previous value of `PDSchedule` will lost forever.

Reply:
> Closed `PDSchedule` and save the previous value in `job.Args`. Now someone killed tidb-server before update `job.RawArgs`. After tidb-server restart, the previous value of `PDSchedule` will lost forever.

Isn't it possible for stage 1, too? Kill server just before the complish of stage 1.

BTW, that is not how it works. If DDL jobs are not persisted, the work will simply lost. You will redo all the work again.

Reply:
> Isn't it possible for stage 1, too? Kill server just before the end of stage 1.
But the `PDSchedule` isn't closed, so we could get the value in next try.

The key problem is close `PDSchedule` is not transactional, we close it by HTTP API. Other operations will rollback when tidb-server be killed before persisted.

Reply:
Got what you mean.